src/wallet/coinselection.cpp

```diff
@@ -384,6 +400,19 @@ CAmount GetSelectionWaste(const std::set<CInputCoin>& inputs, CAmount change_cos
     return waste;
 }
 
+CAmount GenerateChangeTarget(CAmount payment_value)
+{
+    FastRandomContext rng;
```

Comment:
Might be good to use only one rng for coin selection to make mocking and testing easier. Obviously not in this pull, but for now you can at least pass it in to this function?

Reply:
Nice. good first issue?